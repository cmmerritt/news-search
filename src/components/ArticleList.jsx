import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map((article) => (
    <li key={article.url}>
      <Article 
        title={article.title}
        description={article.description}
        author={article.author}
        url={article.url}
        image={article.urlToImage}
      />
    </li>
  ));

  return <ul>{articleElements}</ul>;
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      description: PropTypes.string,
      author: PropTypes.string,
      url: PropTypes.string,
      image: PropTypes.string
    })
  )
};

export default ArticleList;
