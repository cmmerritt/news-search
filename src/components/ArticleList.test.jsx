/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  it('renders ArticleList', () => {
    const { asFragment } = render(
      <ArticleList
        title="Test title"
        description="Test description"
        author="Test author"
        url="testurl.com"
        image="testimg.com"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
