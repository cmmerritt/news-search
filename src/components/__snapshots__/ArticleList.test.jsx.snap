// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <figure>
        <img
          alt="Test title 1"
        />
        <figcaption>
          <p>
            Test title 1
          </p>
          <p>
            Test description 1
          </p>
          <p>
            Test author 1
          </p>
          <p>
            testurl1.com
          </p>
        </figcaption>
      </figure>
    </li>
    <li>
      <figure>
        <img
          alt="Test title 2"
        />
        <figcaption>
          <p>
            Test title 2
          </p>
          <p>
            Test description 2
          </p>
          <p>
            Test author 2
          </p>
          <p>
            testurl2.com
          </p>
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
